src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -139,61 +121,36 @@ def update_stats(self, area, simulation_status, progress_info):
         self.generate_result_report()
 
         self.bids_offers_trades.clear()
+
+        self.trade_profile.update(area)
```

Comment:
Correct me if I am wrong, but this method is called no matter if the csv files / plots are exported or not. This method calculates both global and last slot data irregardless if global data are needed or not. If this is avoided, the corresponding memory leak will be resolved, and we will only keep global data when we need them (simulation is running locally and needs to generate plots)

Reply:
Actually not really. If you look into the function, you will see that it calls `_populate_area_children_data` and this will only calculates global slot data, if self.should_export_plots is False.

Reply:
Awesome. Thanks a lot for explaining!